fetcher.go

```diff
@@ -0,0 +1,5 @@
+package main
+
+type Fetcher interface {
+	Fetch(url string) (body string, urls []string, err error)
```

Comment:
Could you provide some insight on this function?

Reply:
Fetch returns the body of URL and a slice of URLs found on that page.